src/components/PickupBoxForm/PickupBoxForm.jsx

```diff
@@ -0,0 +1,183 @@
+import React, { useState } from 'react';
+import DatePicker from 'react-datepicker';
+import { useForm, Controller } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as yup from 'yup';
+import { PropTypes } from 'prop-types';
+import { FormErrorMessage, FormControl, FormLabel, Input, Button } from '@chakra-ui/react';
+import { formatDate, FYABackend } from '../../common/utils';
+import { uploadBoxPhoto, validateZip } from '../../common/FormUtils/boxFormUtils';
+import DropZone from '../../common/FormUtils/DropZone/DropZone';
+import 'react-datepicker/dist/react-datepicker.css';
+import '../../common/FormUtils/DatePicker.css';
+import styles from './PickupBoxForm.module.css';
+
+yup.addMethod(yup.string, 'isZip', validateZip);
+const schema = yup
+  .object({
+    boxholderName: yup.string().typeError('Invalid name'),
+    boxID: yup
+      .number()
+      .required('Invalid box number, please enter a valid box number')
+      .typeError('Invalid box number, please enter a valid box number'),
+    date: yup
+      .date()
+      .required('Invalid date, please enter a valid date')
+      .typeError('Invalid date, please enter a valid date'),
+    boxholderEmail: yup
+      .string()
+      .required('Invalid email address, please enter a valid email address')
+      .typeError('Invalid email address, please enter a valid email address'),
+    zipcode: yup.string().isZip().required('Invalid zipcode, please enter a valid zipcode'),
+    picture: yup.string().url(),
+  })
+  .required();
+
+const PickupBoxForm = ({ setFormSubmitted }) => {
+  const {
+    register,
+    control,
+    handleSubmit,
+    formState: { errors },
+  } = useForm({
+    resolver: yupResolver(schema),
+    delayError: 750,
+  });
+
+  const [files, setFiles] = useState([]);
+  const [loading, setLoading] = useState(false);
+
+  const onSubmit = async data => {
+    setFormSubmitted(true);
+    const formData = data;
+    formData.date = formatDate(data.date);
+    formData.picture = files.length > 0 ? await uploadBoxPhoto(files[0]) : '';
+
+    try {
+      setLoading(true);
+      await FYABackend.post('/boxHistory', {
+        ...formData,
+        pickup: true,
+        status: 'under review',
+        messageStatus: 'pending',
+        imageStatus: 'pending',
+      });
```

Comment:
The form navigates to the confirmation page even if there is an error with submitting the form through the route. It seems as though the try catch statement is not catching any error and the request sends status error status codes (I test with status code 400) but no error seems to be thrown. This bug is also present in the relocation form.

Reply:
Fixed - this is just a temporary solution. I believe Add Box Form has the same issue so I'll try to look more into it later, but this should be fine for now.